--- STLport/stlport/stl/_ostream.c.orig	2005-10-05 05:28:51.000000000 +0000
+++ STLport/stlport/stl/_ostream.c	2006-10-23 14:52:24.000000000 +0000
@@ -209,7 +209,9 @@
  */
 template <class _CharT, class _Traits>
 basic_ostream<_CharT, _Traits>& basic_ostream<_CharT, _Traits>::operator<<(short __x) {
-  long __tmp = ((this->flags() & _Basic_ios::basefield) != ios_base::dec)?(unsigned short)__x:__x;
+  long __tmp = __x;
+  if ((this->flags() & _Basic_ios::basefield) != ios_base::dec)
+    __tmp = (unsigned short)__x;
   return _M_put_num(*this,  __tmp);
 }
 
@@ -220,7 +222,9 @@
 
 template <class _CharT, class _Traits>
 basic_ostream<_CharT, _Traits>& basic_ostream<_CharT, _Traits>::operator<<(int __x) {
-  long __tmp = ((this->flags() & _Basic_ios::basefield) != ios_base::dec)?(unsigned int)__x:__x;
+  long __tmp = __x;
+  if ((this->flags() & _Basic_ios::basefield) != ios_base::dec)
+    __tmp = (unsigned int)__x;
   return _M_put_num(*this,  __tmp);
 }
 
