--- STLport-5.1.0/stlport/stl/type_manips.h.orig	2006-12-16 20:00:24.000000000 +0000
+++ STLport-5.1.0/stlport/stl/type_manips.h	2006-12-16 20:02:52.000000000 +0000
@@ -38,15 +38,18 @@
  * in gcc 3.x it in bits/type_traits.h that included only from headers
  * that not included from STLport; in gcc 4.x it defined in
  * bits/cpp_type_traits.h, that included well... from cmath for example
+ * in gcc-4.2.0-200612 it was moved to std namespace
  *
  * libstdc++ v3,       __GLIBCXX__ 20050519 (3.4.4) use variant 1,
  * libstdc++ v3,       __GLIBCXX__ 20060306 (3.4.6) use variant 1,
  * while libstdc++ v3, __GLIBCXX__ 20050921 (4.0.2) use variant 2,
- *                     __GLIBCXX__ 20060524 (4.1.1) use variant 2
+ *                     __GLIBCXX__ 20060524 (4.1.1) use variant 2,
+ *                     __GLIBCXX__ 20061119 (4.2.0) use variant 2,
+ *                     __GLIBCXX__ 20061206 (4.2.0) use variant 1
  * muddle in libstdc++ versions...
  *
  */
-# if defined (__GNUC__) && (__GNUC__ > 3)
+# if defined (__GNUC__) && (__GNUC__ > 3) && defined (__GNUC_MINOR__) && (__GNUC_MINOR__ < 2)
 using ::__true_type;
 using ::__false_type;
 # else
