--- STLport-4.5.3/src/gcc-linux.mak.extra-cxxflags	2002-01-18 23:06:52.000000000 +0100
+++ STLport-4.5.3/src/gcc-linux.mak	2002-08-20 10:38:56.000000000 +0200
@@ -2,7 +2,7 @@
 # compiler
 #
 CC = gcc
-CXX = c++  -pthread -fexceptions
+CXX = g++
 
 #
 # Basename for libraries
@@ -14,7 +14,7 @@ LIB_BASENAME = libstlport_gcc
 #
 #
 LINK=ar crv
-DYN_LINK=gcc -shared -o
+DYN_LINK=$(CXX) -lpthread -shared -o
 
 OBJEXT=o
 DYNEXT=so
@@ -38,10 +38,10 @@ WARNING_FLAGS= -W -Wno-sign-compare -Wno
 
 # boris : real locale implementation may not really work
 # CXXFLAGS_COMMON = -I${STLPORT_DIR} ${WARNING_FLAGS} -D_STLP_REAL_LOCALE_IMPLEMENTED
-CXXFLAGS_COMMON = -I${STLPORT_DIR} ${WARNING_FLAGS}
+CXXFLAGS_COMMON = -I${STLPORT_DIR} ${WARNING_FLAGS} -pthread
 
-CXXFLAGS_RELEASE_static = $(CXXFLAGS_COMMON) -O2
-CXXFLAGS_RELEASE_dynamic = $(CXXFLAGS_COMMON) -O2 -fPIC
+CXXFLAGS_RELEASE_static = $(CXXFLAGS_COMMON) -O2 $(EXTRA_CXXFLAGS)
+CXXFLAGS_RELEASE_dynamic = $(CXXFLAGS_COMMON) -O2 $(EXTRA_CXXFLAGS) -fPIC
 
 CXXFLAGS_DEBUG_static = $(CXXFLAGS_COMMON) -O -g
 CXXFLAGS_DEBUG_dynamic = $(CXXFLAGS_COMMON) -O -g -fPIC
