diff -Nur STLport-4.5.1/src/common_macros.mak STLport-4.5.1.new/src/common_macros.mak
--- STLport-4.5.1/src/common_macros.mak	Thu Nov 15 23:08:56 2001
+++ STLport-4.5.1.new/src/common_macros.mak	Sat Dec 15 22:45:50 2001
@@ -29,14 +29,8 @@
 OBJDIR=$(OUTDIR)$(PATH_SEP)obj$(PATH_SEP)$(COMP)
 
 RELEASE_OBJDIR_static=$(OBJDIR)$(PATH_SEP)Release
-DEBUG_OBJDIR_static=$(OBJDIR)$(PATH_SEP)Debug
-STLDEBUG_OBJDIR_static=$(OBJDIR)$(PATH_SEP)DebugSTL
 RELEASE_OBJDIR_staticx=$(OBJDIR)$(PATH_SEP)Releasex
-DEBUG_OBJDIR_staticx=$(OBJDIR)$(PATH_SEP)Debugx
-STLDEBUG_OBJDIR_staticx=$(OBJDIR)$(PATH_SEP)DebugSTLx
 RELEASE_OBJDIR_dynamic=$(OBJDIR)$(PATH_SEP)ReleaseD
-DEBUG_OBJDIR_dynamic=$(OBJDIR)$(PATH_SEP)DebugD
-STLDEBUG_OBJDIR_dynamic=$(OBJDIR)$(PATH_SEP)DebugSTLD
 
 #
 # By default on UNIX, STLport headers go into /usr/local/include/stlport,
@@ -59,33 +53,20 @@
 #
 
 RELEASE_NAME=$(LIB_BASENAME)
-DEBUG_NAME=$(LIB_BASENAME)_debug
-STLDEBUG_NAME=$(LIB_BASENAME)_stldebug
 
 RELEASE_DYNLIB=$(RELEASE_NAME).$(DYNEXT)$(DYNAMIC_SUFFIX)
-DEBUG_DYNLIB=$(DEBUG_NAME).$(DYNEXT)$(DYNAMIC_SUFFIX)
-STLDEBUG_DYNLIB=$(STLDEBUG_NAME).$(DYNEXT)$(DYNAMIC_SUFFIX)
 
 RELEASE_DYNLIB_SONAME=$(RELEASE_NAME).$(DYNEXT)$(VERSION_SUFFIX)
-DEBUG_DYNLIB_SONAME=$(DEBUG_NAME).$(DYNEXT)$(VERSION_SUFFIX)
-STLDEBUG_DYNLIB_SONAME=$(STLDEBUG_NAME).$(DYNEXT)$(VERSION_SUFFIX)
 
 RELEASE_LIB=$(RELEASE_NAME)$(STATIC_SUFFIX).$(STEXT)
-DEBUG_LIB=$(DEBUG_NAME)$(STATIC_SUFFIX).$(STEXT)
-STLDEBUG_LIB=$(STLDEBUG_NAME)$(STATIC_SUFFIX).$(STEXT)
 
 RELEASEX_LIB=$(RELEASE_NAME)$(STATIC_SUFFIX)x.$(STEXT)
-DEBUGX_LIB=$(DEBUG_NAME)$(STATIC_SUFFIX)x.$(STEXT)
-STLDEBUGX_LIB=$(STLDEBUG_NAME)$(STATIC_SUFFIX)x.$(STEXT)
 
 ALL_STATIC_LIBS=$(OUTDIR)$(PATH_SEP)$(RELEASE_LIB) $(OUTDIR)$(PATH_SEP)$(STLDEBUG_LIB)
-DEBUG_STATIC_LIBS=$(ALL_STATIC_LIBS) $(OUTDIR)$(PATH_SEP)$(DEBUG_LIB)
 
 ALL_STATICX_LIBS=$(OUTDIR)$(PATH_SEP)$(RELEASEX_LIB) $(OUTDIR)$(PATH_SEP)$(STLDEBUGX_LIB)
-DEBUG_STATICX_LIBS=$(ALL_STATICX_LIBS) $(OUTDIR)$(PATH_SEP)$(DEBUGX_LIB)
 
 ALL_DYNAMIC_LIBS=$(OUTDIR)$(PATH_SEP)$(RELEASE_DYNLIB) $(OUTDIR)$(PATH_SEP)$(STLDEBUG_DYNLIB)
-DEBUG_DYNAMIC_LIBS=$(OUTDIR)$(PATH_SEP)$(RELEASE_DYNLIB) $(OUTDIR)$(PATH_SEP)$(STLDEBUG_DYNLIB)
 
 
 RELEASE_OBJECTS_static= \
@@ -124,76 +105,6 @@
 $(RELEASE_OBJDIR_static)$(PATH_SEP)string_w.$(OBJEXT)
 
 
-DEBUG_OBJECTS_static= \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)dll_main.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)fstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)strstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)sstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)ios.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)streambuf.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)stdio_streambuf.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)istream.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)ostream.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)iostream.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)codecvt.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)collate.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)ctype.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)monetary.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)num_get.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)num_put.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)num_get_float.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)num_put_float.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)numpunct.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)time_facets.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)messages.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)locale_impl.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)locale.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)locale_catalog.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)facets_byname.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)c_locale.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)c_locale_stub.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)complex.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)complex_exp.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)complex_io.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)complex_trig.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)complex_io_w.$(OBJEXT) \
-$(DEBUG_OBJDIR_static)$(PATH_SEP)string_w.$(OBJEXT)
-
-STLDEBUG_OBJECTS_static= \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)dll_main.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)fstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)strstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)sstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)ios.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)streambuf.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)stdio_streambuf.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)istream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)ostream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)iostream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)codecvt.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)collate.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)ctype.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)monetary.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)num_get.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)num_put.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)num_get_float.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)num_put_float.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)numpunct.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)time_facets.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)messages.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)locale_impl.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)locale.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)locale_catalog.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)facets_byname.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)c_locale.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)c_locale_stub.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)complex.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)complex_exp.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)complex_io.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)complex_trig.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)complex_io_w.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_static)$(PATH_SEP)string_w.$(OBJEXT)
-
 ###########################################################
 
 RELEASE_OBJECTS_staticx= \
@@ -232,77 +143,6 @@
 $(RELEASE_OBJDIR_staticx)$(PATH_SEP)string_w.$(OBJEXT)
 
 
-DEBUG_OBJECTS_staticx= \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)dll_main.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)fstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)strstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)sstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)ios.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)streambuf.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)stdio_streambuf.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)istream.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)ostream.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)iostream.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)codecvt.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)collate.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)ctype.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)monetary.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)num_get.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)num_put.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)num_get_float.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)num_put_float.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)numpunct.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)time_facets.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)messages.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)locale_impl.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)locale.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)locale_catalog.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)facets_byname.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)c_locale.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)c_locale_stub.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)complex.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)complex_exp.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)complex_io.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)complex_trig.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)complex_io_w.$(OBJEXT) \
-$(DEBUG_OBJDIR_staticx)$(PATH_SEP)string_w.$(OBJEXT)
-
-STLDEBUG_OBJECTS_staticx= \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)dll_main.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)fstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)strstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)sstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)ios.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)streambuf.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)stdio_streambuf.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)istream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)ostream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)iostream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)codecvt.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)collate.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)ctype.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)monetary.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)num_get.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)num_put.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)num_get_float.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)num_put_float.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)numpunct.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)time_facets.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)messages.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)locale_impl.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)locale.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)locale_catalog.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)facets_byname.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)c_locale.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)c_locale_stub.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)complex.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)complex_exp.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)complex_io.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)complex_trig.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)complex_io_w.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_staticx)$(PATH_SEP)string_w.$(OBJEXT)
-
-
 ###########################################################
 
 RELEASE_OBJECTS_dynamic= \
@@ -341,76 +181,4 @@
 $(RELEASE_OBJDIR_dynamic)$(PATH_SEP)string_w.$(OBJEXT) \
 $(RESFILE)
 
-DEBUG_OBJECTS_dynamic= \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)dll_main.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)fstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)strstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)sstream.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)ios.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)streambuf.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)stdio_streambuf.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)istream.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)ostream.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)iostream.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)codecvt.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)collate.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)ctype.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)monetary.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)num_get.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)num_put.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)num_get_float.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)num_put_float.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)numpunct.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)time_facets.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)messages.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)locale_impl.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)locale.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)locale_catalog.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)facets_byname.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)c_locale.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)c_locale_stub.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)complex.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_exp.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_io.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_trig.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_io_w.$(OBJEXT) \
-$(DEBUG_OBJDIR_dynamic)$(PATH_SEP)string_w.$(OBJEXT) \
-$(RESFILE_debug)
-
-STLDEBUG_OBJECTS_dynamic= \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)dll_main.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)fstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)strstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)sstream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)ios.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)streambuf.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)stdio_streambuf.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)istream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)ostream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)iostream.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)codecvt.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)collate.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)ctype.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)monetary.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)num_get.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)num_put.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)num_get_float.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)num_put_float.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)numpunct.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)time_facets.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)messages.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)locale_impl.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)locale.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)locale_catalog.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)facets_byname.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)c_locale.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)c_locale_stub.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)complex.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_exp.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_io.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_trig.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)complex_io_w.$(OBJEXT) \
-$(STLDEBUG_OBJDIR_dynamic)$(PATH_SEP)string_w.$(OBJEXT) \
-$(RESFILE_stldebug)
-
 
diff -Nur STLport-4.5.1/src/common_rules.mak STLport-4.5.1.new/src/common_rules.mak
--- STLport-4.5.1/src/common_rules.mak	Mon Oct  8 19:30:30 2001
+++ STLport-4.5.1.new/src/common_rules.mak	Sat Dec 15 23:04:26 2001
@@ -81,11 +81,7 @@
 
 symbolic_links :
 	$(RM) $(OUTDIR)/$(RELEASE_NAME).$(DYNEXT)
-	$(RM) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
-	$(RM) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
 	ln -s $(RELEASE_DYNLIB) $(OUTDIR)/$(RELEASE_NAME).$(DYNEXT)
-	-ln -s $(DEBUG_DYNLIB) $(OUTDIR)/$(DEBUG_NAME).$(DYNEXT)
-	ln -s $(STLDEBUG_DYNLIB) $(OUTDIR)/$(STLDEBUG_NAME).$(DYNEXT)
 
 install_unix :
 	-$(RM) $(INSTALLDIR_INC)
@@ -112,9 +108,7 @@
 	 $(INSTALL_LIB)  $$file $(INSTALLDIR_LIB); \
 	done
 	$(RM) $(INSTALLDIR_LIB)/$(RELEASE_NAME).$(DYNEXT)
-	$(RM) $(INSTALLDIR_LIB)/$(STLDEBUG_NAME).$(DYNEXT)
 	ln -s $(RELEASE_DYNLIB) $(INSTALLDIR_LIB)/$(RELEASE_NAME).$(DYNEXT)
-	ln -s $(STLDEBUG_DYNLIB) $(INSTALLDIR_LIB)/$(STLDEBUG_NAME).$(DYNEXT)
 
 #	ln -s $(INSTALLDIR_LIB)/$(RELEASE_DYNLIB) $(INSTALLDIR_LIB)/$(RELEASE_NAME).$(DYNEXT)
 #	ln -s $(INSTALLDIR_LIB)/$(STLDEBUG_DYNLIB) $(INSTALLDIR_LIB)/$(STLDEBUG_NAME).$(DYNEXT)
@@ -134,24 +128,6 @@
 $(OUTDIR)$(PATH_SEP)$(RELEASEX_LIB) : $(OUTDIR) $(RELEASE_OBJDIR_staticx) $(DEF_FILE) $(RELEASE_OBJECTS_staticx)
 	$(LINK) $(LINK_OUT)$(OUTDIR)$(PATH_SEP)$(RELEASEX_LIB) $(LDFLAGS_RELEASE_static) $(RELEASE_OBJECTS_staticx) $(LDLIBS_RELEASE_static)
 
-$(OUTDIR)$(PATH_SEP)$(DEBUG_DYNLIB) : $(OUTDIR) $(DEBUG_OBJDIR_dynamic) $(DEF_FILE) $(DEBUG_OBJECTS_dynamic)
-	$(DYN_LINK) $(DYNLINK_OUT)$(OUTDIR)$(PATH_SEP)$(DEBUG_DYNLIB) $(LDFLAGS_DEBUG_dynamic) $(DEBUG_OBJECTS_dynamic) $(LDLIBS_DEBUG_dynamic) 
-
-$(OUTDIR)$(PATH_SEP)$(DEBUG_LIB) : $(OUTDIR) $(DEBUG_OBJDIR_static) $(DEF_FILE) $(DEBUG_OBJECTS_static)
-	$(LINK) $(LINK_OUT)$(OUTDIR)$(PATH_SEP)$(DEBUG_LIB) $(LDLIBS_DEBUG_static) $(LDFLAGS_DEBUG_static) $(DEBUG_OBJECTS_static)  
-
-$(OUTDIR)$(PATH_SEP)$(DEBUGX_LIB) : $(OUTDIR) $(DEBUG_OBJDIR_staticx) $(DEF_FILE) $(DEBUG_OBJECTS_staticx)
-	$(LINK) $(LINK_OUT)$(OUTDIR)$(PATH_SEP)$(DEBUGX_LIB) $(LDLIBS_DEBUG_static) $(LDFLAGS_DEBUG_static) $(DEBUG_OBJECTS_staticx)
-
-$(OUTDIR)$(PATH_SEP)$(STLDEBUG_DYNLIB) : $(OUTDIR) $(STLDEBUG_OBJDIR_dynamic) $(DEF_FILE) $(STLDEBUG_OBJECTS_dynamic)
-	$(DYN_LINK) $(DYNLINK_OUT)$(OUTDIR)$(PATH_SEP)$(STLDEBUG_DYNLIB) $(LDFLAGS_STLDEBUG_dynamic) $(STLDEBUG_OBJECTS_dynamic) $(LDLIBS_STLDEBUG_dynamic)
-
-$(OUTDIR)$(PATH_SEP)$(STLDEBUG_LIB) : $(OUTDIR) $(STLDEBUG_OBJDIR_static) $(DEF_FILE) $(STLDEBUG_OBJECTS_static)
-	$(LINK) $(LINK_OUT)$(OUTDIR)$(PATH_SEP)$(STLDEBUG_LIB) $(LDFLAGS_STLDEBUG_static) $(STLDEBUG_OBJECTS_static)  $(LDLIBS_STLDEBUG_static)
-
-$(OUTDIR)$(PATH_SEP)$(STLDEBUGX_LIB) : $(OUTDIR) $(STLDEBUG_OBJDIR_staticx) $(DEF_FILE) $(STLDEBUG_OBJECTS_staticx)
-	$(LINK) $(LINK_OUT)$(OUTDIR)$(PATH_SEP)$(STLDEBUGX_LIB) $(LDFLAGS_STLDEBUG_static) $(STLDEBUG_OBJECTS_staticx)  $(LDLIBS_STLDEBUG_static)
-
 # locale_impl.cpp : codecvt.cpp collate.cpp ctype.cpp monetary.cpp num_get.cpp num_get_float.cpp num_put.cpp num_put_float.cpp numpunct.cpp time_facets.cpp messages.cpp
 
 # locale_catalog.cpp : ctype_byname.cpp collate_byname.cpp codecvt_byname.cpp numpunct_byname.cpp monetary_byname.cpp messages_byname.cpp
diff -Nur STLport-4.5.1/src/gcc.mak STLport-4.5.1.new/src/gcc.mak
--- STLport-4.5.1/src/gcc.mak	Sat Dec  1 22:00:18 2001
+++ STLport-4.5.1.new/src/gcc.mak	Sat Dec 15 22:48:31 2001
@@ -38,14 +38,8 @@
 
 CXXFLAGS_COMMON = -I${STLPORT_DIR} ${WARNING_FLAGS}
 
-CXXFLAGS_RELEASE_static = $(CXXFLAGS_COMMON) -O2
-CXXFLAGS_RELEASE_dynamic = $(CXXFLAGS_COMMON) -O2 -fPIC
-
-CXXFLAGS_DEBUG_static = $(CXXFLAGS_COMMON) -O -g
-CXXFLAGS_DEBUG_dynamic = $(CXXFLAGS_COMMON) -O -g -fPIC
-
-CXXFLAGS_STLDEBUG_static = $(CXXFLAGS_DEBUG_static) -D_STLP_DEBUG
-CXXFLAGS_STLDEBUG_dynamic = $(CXXFLAGS_DEBUG_dynamic) -D_STLP_DEBUG
+CXXFLAGS_RELEASE_static = $(CXXFLAGS_COMMON) $(CXXFLAGS)
+CXXFLAGS_RELEASE_dynamic = $(CXXFLAGS_COMMON) $(CXXFLAGS) -fPIC
 
 include common_percent_rules.mak
 include common_rules.mak
